src/features/search/pages/DatesHoursModal.tsx

```diff
@@ -0,0 +1,366 @@
+import { SearchFixedModalBottom } from 'features/search/components/SearchFixedModalBottom'
+import { DATE_FILTER_OPTIONS } from 'features/search/enums'
+import { datesHoursSchema } from 'features/search/pages/schema/datesHoursSchema'
+import { useSearch } from 'features/search/pages/SearchWrapper'
+import { useGetFullscreenModalSliderLength } from 'features/search/pages/useGetFullscreenModalSliderLength'
+import { SectionTitle } from 'features/search/sections/titles'
+import { SearchState, SearchView } from 'features/search/types'
+import { useLogFilterOnce } from 'features/search/utils/useLogFilterOnce'
+import { formatToCompleteFrenchDate } from 'libs/parsers'
+import { Range } from 'libs/typesUtils/typeHelpers'
+import { Form } from 'ui/components/Form'
+import { Slider } from 'ui/components/inputs/Slider'
+import { Li } from 'ui/components/Li'
+import { AppModal } from 'ui/components/modals/AppModal'
+import { RadioButton } from 'ui/components/radioButtons/RadioButton'
+import { Separator } from 'ui/components/Separator'
+import { VerticalUl } from 'ui/components/Ul'
+import { Close } from 'ui/svg/icons/Close'
+import { Spacer, Typo } from 'ui/theme'
+
+export enum RadioButtonDate {
+  TODAY = 'Aujourd’hui',
+  WEEK = 'Cette semaine',
+  WEEK_END = 'Ce week-end',
+  PRECISE_DATE = 'Date précise',
+}
+
+type DatesHoursModalFormData = {
+  hasSelectedDate: boolean
+  selectedDateChoice: DATE_FILTER_OPTIONS
+  selectedDate?: Date
+  hasSelectedHours: boolean
+  selectedHours?: Range<number>
+}
+
+type Props = {
+  title: string
+  accessibilityLabel: string
+  isVisible: boolean
+  hideModal: () => void
+}
+
+export const DATE_TYPES: Array<{
+  label: RadioButtonDate
+  type: DATE_FILTER_OPTIONS
+}> = [
+  { label: RadioButtonDate.TODAY, type: DATE_FILTER_OPTIONS.TODAY },
+  { label: RadioButtonDate.WEEK, type: DATE_FILTER_OPTIONS.CURRENT_WEEK },
+  { label: RadioButtonDate.WEEK_END, type: DATE_FILTER_OPTIONS.CURRENT_WEEK_END },
+  { label: RadioButtonDate.PRECISE_DATE, type: DATE_FILTER_OPTIONS.USER_PICK },
+]
+
+const MAX_HOUR = 24
+
+const DEFAULT_TIME_RANGE: Range<number> = [8, 22]
+
+const formatHour = (hour: number) => `${hour}h`
+
+const titleId = uuidv4()
+
+export const DatesHoursModal: FunctionComponent<Props> = ({
+  title,
+  accessibilityLabel,
+  isVisible,
+  hideModal,
+}) => {
+  const { navigate } = useNavigation<UseNavigationType>()
+  const { isDesktopViewport, modal } = useTheme()
+  const { searchState } = useSearch()
+  const logToggleDate = useLogFilterOnce(SectionTitle.Date)
+  const logToggleHour = useLogFilterOnce(SectionTitle.Hour)
+  const logSelectedDateChoice = useLogFilterOnce(SectionTitle.OfferDate)
+  const logChangeSelectedHours = useLogFilterOnce(SectionTitle.TimeSlot)
+  const [showCalendarPicker, setShowCalendarPicker] = useState<boolean>(false)
+  const { sliderLength } = useGetFullscreenModalSliderLength()
+
+  const TODAY = useMemo(() => new Date(), [])
```

Comment:
Ugh, je pense que c'est vraiment pas nécessaire de mettre ça dans un `useMemo`.
Tu peux le sortir tu composant, et avoir simplement un `const TODAY = new Date()`

Reply:
J'ai fait ça par rapport aux tests unitaires pour que ça soit ok avec l'utilisation de `mockdate`.
En le sortant du composant le forçage de la date n'était pas ok.

Reply:
Ah y'a un outil externe pour mocker une date ?
Il me semble que j'utilisais simplement un truc du genre `global.Date =` ou un truc du genre.

Quoiqu'il en soit c'est pas fou je trouve... C'est commun dans le projet ?

Reply:
Yes `mockdate` est commun